### PR TITLE
Gradle and Spring upgrade in Mock Spine Mhs Outbound

### PR DESCRIPTION
**Pull request by ole4ryb**
## What

Gradle and Spring boot version upgrade

## Why

The Mock Spine MHS Outbound module in the test suite was using outdated Gradle and Spring Boot versions, which caused it to lag behind other modules. This PR updates the module to align with the versions used by other components, ensuring compatibility across Java versions and preventing any potential issues related to version mismatches.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Internal change (non-breaking change with no effect on the functionality affecting end users)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have updated the [Changelog](/CHANGELOG.md) with details of my change in the UNRELEASED section if this change will affect end users
- [ ] A corresponding change has been made to the [Mapping Documentation repository][mapping-docs]

[mapping-docs]: https://github.com/NHSDigital/patient-switching-adaptors-mapping-documentation

### CODE REVIEW

File: test-suite/mock-spine-mhs-outbound/build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 plugins {
-    id 'org.springframework.boot' version '2.7.1'
-    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
-    id 'io.freefair.lombok' version '6.2.0'
+    id 'org.springframework.boot' version '3.3.3'
```
Comment (martin-nhs) [inline test-suite/mock-spine-mhs-outbound/build.gradle:2]: Wouldn't it be better if we allow the patch level to be the latest, leaving the major and minor levels as-is? E.g. `3.3.x`?

Comment (ole4ryb) [reply test-suite/mock-spine-mhs-outbound/build.gradle:2]: I just took it from other project services where we are using minor version which currently is ver. 3.3.3,

File: test-suite/mock-spine-mhs-outbound/build.gradle [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+    id 'io.spring.dependency-management' version '1.1.6'
```
Comment (martin-nhs) [inline test-suite/mock-spine-mhs-outbound/build.gradle:3]: Same as the comment above, `1.1` should suffice - if we choose that this is OK to do so.

Comment (ole4ryb) [reply test-suite/mock-spine-mhs-outbound/build.gradle:3]: again, I am aligning it with the other modules, like gp2gp-translator or common or db-connector... they are all using ver. 1.1.6. I think the general rule is that it is better to use more specific version given we are already have other modules using the same version.

File: test-suite/mock-spine-mhs-outbound/build.gradle (hunk 1 above)
Comment (martin-nhs) [reply test-suite/mock-spine-mhs-outbound/build.gradle:2]: OK, no biggie - just thought I would highlight it 😄

File: test-suite/mock-spine-mhs-outbound/build.gradle (hunk 2 above)
Comment (martin-nhs) [reply test-suite/mock-spine-mhs-outbound/build.gradle:3]: Okay doak! Makes sense, just thought I'd highlight!

**Review by martin-nhs — APPROVED** [review]

**Pull request merged**